sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetPartitionDiscoverySuite.scala

```diff
@@ -101,11 +118,13 @@ class ParquetPartitionDiscoverySuite extends QueryTest with ParquetTest with Sha
 
     checkThrows[AssertionError]("file://path/=10", "Empty partition column name")
     checkThrows[AssertionError]("file://path/a=", "Empty partition column value")
+    checkThrows[AssertionError]("file://path/a=b=c", "Not a partition format in")
```

Comment:
You're right, it's not related for this PR, but a very trivial checking, with more informative message for the invalid partition in the path.